python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,509 @@
+    sqlite database.
+
+    Create a simple sqlite database and implement wrappers to store and
+    retrieve DIAObjects and DIASources from within that database. This task
+    functions as a testing ground for the L1 database and should mimic this
+    database's eventual functionaly. This specific database implementation is
+    useful for the verification packages which may not be run with access to
+    L1 database.
+
+    Attriburtes
+    -----------
+    indexer : lsst.meas.algorithms.IndexerRegistry
+        An spatial indexing object for fast look up of stored DIAObjects.
+    """
+
+    ConfigClass = AssociationDBSqliteConfig
+    _DefaultName = "association_db_sqlite"
+
+    def __init__(self, **kwargs):
+
+        pipeBase.Task.__init__(self, **kwargs)
+        self.indexer = IndexerRegistry[self.config.indexer.name](
+            self.config.indexer.active)
+        self._db_connection = sqlite3.connect(self.config.db_name)
+        self._db_cursor = self._db_connection.cursor()
+
+        self._dia_object_converter = SqliteDBConverter(
+            make_minimal_dia_object_schema(), "dia_objects")
+        self._dia_source_converter = SqliteDBConverter(
+            make_minimal_dia_source_schema(), "dia_sources")
+
+    def run(self):
+        """ This subtask does not make use of the run method for Tasks.
+
+        Please use the store or load methods.
+        """
+        return None
+
+    def _commit(self):
+        """ Save changes to the sqlite database.
+        """
+        self._db_connection.commit()
+
+    def close(self):
+        """ Close the connection to the sqlite database.
+        """
+        self._db_connection.close()
+
+    def create_tables(self):
+        """ If no sqlite database exists with the correct tables we can create
+        one using this method.
+
+        Returns
+        -------
+        bool
+            Successfully created a new database with specified tables.
+        """
+
+        self._db_cursor.execute(
+            'select name from sqlite_master where type = "table"')
+        db_tables = self._db_cursor.fetchall()
+
+        # If this database currently contains any tables exit and do not
+        # create tables.
+        if db_tables:
+            return False
+        else:
+            # Create databases to store the individual DIAObjects and DIASources
+            self._db_cursor.execute(
+                self._dia_object_converter.make_table_from_afw_shcema(
+                    "dia_objects"))
+            self._commit()
+            self._db_cursor.execute(
+                self._dia_source_converter.make_table_from_afw_shcema(
+                    "dia_sources"))
+            self._commit()
+
+            # Create linkage database between associated dia_objects and
+            # dia_sources.
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects_to_dia_sources ("
+                "src_id INTEGER, "
+                "obj_id INTEGER, "
```

Comment:
I noticed you got rid of the primary key entirely. Why?

Reply:
This was an oversight on my part when I added the FOREIGN KEY constraint. I have added the PRIMARY KEY back.